Lost active contexts when using static weld

When I using `static` weld object for creating `WeldInitiator`, active contexts was lost after two tests.
```
[ERROR] Errors: 
[ERROR]   WeldTest.doTest3:42 » ContextNotActive WELD-001303: No active contexts for scope type javax.enterprise.context.RequestScoped
[ERROR]   WeldTest.doTest4:47 » ContextNotActive WELD-001303: No active contexts for scope type javax.enterprise.context.RequestScoped
``` 
But when I using `WeldInitiator.from(<classes>)` all tests works fine.

The version of weld-junit is 2.0.2.Final, weld is 3.1.9 (Final), junit is 5.4.2.

```java
@RequestScoped
class ServiceFoo {
    void run() {
    }
}

@ExtendWith(WeldJunit5Extension.class)
class WeldTest {

    @Inject
    protected ServiceFoo serviceFoo;

    static final Weld weld = WeldInitiator.createWeld();
    static {
        weld.addBeanClass(ServiceFoo.class);
    }

    @WeldSetup
    public WeldInitiator weldInitiator =
            WeldInitiator.from(weld)
            //WeldInitiator.from(ServiceFoo.class)
                    .activate(RequestScoped.class)
                    .build();

    @Test
    void doTest1() {
        serviceFoo.run();
    }

    @Test
    void doTest2() {
        serviceFoo.run();
    }

    @Test
    void doTest3() {
        serviceFoo.run();
    }

    @Test
    void doTest4() {
        serviceFoo.run();
    }
}
```

--- a/junit5/src/main/java/org/jboss/weld/junit5/WeldInitiator.java
+++ b/junit5/src/main/java/org/jboss/weld/junit5/WeldInitiator.java
@@ -70,8 +70,13 @@ public class WeldInitiator extends AbstractWeldInitiator {
 
     /**
      * The container is configured through a provided {@link Weld} instance.
+     * <p>
+     * Provided instance of {@link Weld} should be kept in the same scope as the resulting {@link WeldInitiator}.
+     * I.e. if {@link Weld} instance is kept in a static field, so should be the resulting {@link WeldInitiator}.
+     * Failing to uphold this can lead to repetitive configuration of the same {@link Weld} instance which in turn results in
+     * unexpected container setup/behavior.
      *
-     * @param weld
+     * @param weld instance of {@link Weld} used to create {@link WeldInitiator}
      * @return a new WeldInitiator instance
      */
     public static WeldInitiator of(Weld weld) {
@@ -120,8 +125,13 @@ public class WeldInitiator extends AbstractWeldInitiator {
 
     /**
      * Create a builder instance.
+     * <p>
+     * Provided instance of {@link Weld} should be kept in the same scope as the resulting {@link WeldInitiator}.
+     * I.e. if {@link Weld} instance is kept in a static field, so should be the resulting {@link WeldInitiator}.
+     * Failing to uphold this can lead to repetitive configuration of the same {@link Weld} instance which in turn results in
+     * unexpected container setup/behavior.
      *
-     * @param weld
+     * @param weld instance of {@link Weld} used as a basis for this {@link WeldInitiator.Builder}
      * @return a builder instance
      * @see #of(Weld)
      */

